src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -435,6 +435,14 @@ static bool TryParseValue(string value, out string result)
             Compilation compilation)
             => options.GetSymbolNamesWithValueOption<Unit>(EditorConfigOptionNames.LinqChainMethods, rule, tree, compilation, static name => new SymbolNamesWithValueOption<Unit>.NameParts(name, Unit.Default), namePrefix: "M:");
 
+        public static bool GetAssumeMethodEnumeratesArgumentsOption(
+            this AnalyzerOptions options,
+            DiagnosticDescriptor rule,
+            SyntaxTree tree,
+            Compilation compilation,
+            bool defaultValue)
```

Comment:
Do we expect any callers to have a different default value than `false`? Otherwise, we should remove this parameter and just hard code `defaultValue: false` in the below call. I am also fine if you want to inline the `options.GetBoolOptionValue` call directly into the analyzer and avoid adding this new extension method `GetAssumeMethodEnumeratesArgumentsOption` completely.